src/utils/middlewares/__tests__/azure_api_auth.test.ts

```diff
@@ -256,3 +262,107 @@ describe("AzureApiAuthMiddleware", () => {
     }
   });
 });
+
+describe("AzureAllowBodyPayloadMiddleware", () => {
+  it("should success if pattern is not matched", async () => {
+    const headers = {
+      ...someHeaders
+    };
+    const aPayload = { foo: { bar: "baz" } };
+    const mockRequest = {
+      header: jest.fn(lookup(headers)),
+      body: aPayload
+    };
+    const aNonMatchingCodec = t.interface({ anyField: t.number });
+    const anyGroupSet = new Set([UserGroup.ApiMessageWrite]);
+
+    const middleware = AzureAllowBodyPayloadMiddleware(
+      aNonMatchingCodec,
+      anyGroupSet
+    );
+
+    const result = await middleware(mockRequest as any);
+
+    expect(isRight(result)).toBe(true);
+    expect(aNonMatchingCodec.decode(aPayload).isLeft()).toBe(true); // test is wrong if it fails
+  });
+
+  it("should success if pattern is not matched", async () => {
+    const headers = {
+      ...someHeaders
+    };
+    const aPayload = { foo: { bar: "baz" } };
+    const mockRequest = {
+      header: jest.fn(lookup(headers)),
+      body: aPayload
+    };
+    const aNonMatchingCodec = t.interface({ anyField: t.string });
+    const anyGroupSet = new Set([UserGroup.ApiDebugRead]); // any value
+
+    const middleware = AzureAllowBodyPayloadMiddleware(
+      aNonMatchingCodec,
+      anyGroupSet
+    );
+
+    const result = await middleware(mockRequest as any);
+
+    expect(isRight(result)).toBe(true);
+    expect(aNonMatchingCodec.decode(aPayload).isLeft()).toBe(true); // test is wrong if it fails
+  });
+
```

Comment:
Add test that pass if patter is matched and user belongs to an allowed group

```suggestion

  it("should success if pattern is matched and current user belongs to user group", async () => {
    const headers = {
      ...someHeaders
    };
    const aPayload = { foo: { bar: "baz" } };
    const mockRequest = {
      header: jest.fn(lookup(headers)),
      body: aPayload
    };
    const aMatchingCodec = t.interface({
      foo: t.interface({ bar: t.string })
    });
    const anyGroupSet = new Set([UserGroup.ApiMessageWrite]);

    const middleware = AzureAllowBodyPayloadMiddleware(
      aMatchingCodec,
      anyGroupSet
    );

    const result = await middleware(mockRequest as any);

    expect(isRight(result)).toBe(true);
    expect(aMatchingCodec.decode(aPayload).isRight()).toBe(true); // test is wrong if it fails
  });

```

Reply:
Shouldn't be:
```
   const headers = {
      ...someHeaders,
      "x-user-groups": "ApiMessageWrite"
    };
```
?

Reply:
`someHeaders` already contains `"x-user-groups": "ApiMessageWrite"`

Reply:
Ok, good.